modular_skyrat/master_files/code/modules/clothing/towels.dm

```diff
@@ -0,0 +1,523 @@
+		return
+
+	if(!ishuman(user) || !iscyborg(user))
+		if(iscyborg(user) && wet) // Cyborgs can't wring towels.
+			to_chat(user, span_warning("Folding a wet towel doesn't really make sense. You stop yourself before doing that."))
+			return
+
+		var/in_hands = TRUE
+
+		if(ishuman(user))
+			in_hands = user.get_active_held_item() == src || user.get_inactive_held_item() == src
+
+
+		if(!in_hands) // They need to be in your hands, unless you're a cyborg.
+			return
+
+		if(!wet)
+			change_towel_shape(user, TOWEL_FOLDED, silent = TRUE)
+			to_chat(user, span_notice("You fold [src] up neatly."))
+			return
+
+		// No cyborgs past this point.
+
+		to_chat(user, span_warning("You start wringing [src], it's going to make a mess!"))
+
+		if(!do_after(user, 2 SECONDS, src))
+			to_chat(user, span_warning("You give wringing [src] a second thought, and stop doing it, maybe for the best..."))
+			return
+
+		var/turf/current_turf = get_turf(src) // It's done by a user so it should always have a turf.
+
+		var/datum/reagents/tempr = new(max_reagent_volume)
+		var/transfer_amount = min(reagents.total_volume, TOWEL_WRING_AMOUNT)
+
+		transfer_towel_reagents_to(tempr, transfer_amount, user, loss_factor = TOWEL_WRING_LOSS_FACTOR, make_used = TRUE)
+
+		current_turf.add_liquid_from_reagents(tempr)
+
+		qdel(tempr)
+
+		user.visible_message(span_warning("[user] wrings [src], making a mess on \the [current_turf]!"), span_warning("You wring [src], making a mess on \the [current_turf]!"))
+
+
+/obj/item/towel/machine_wash(obj/machinery/washing_machine/washer)
+	. = ..() // This isn't really needed, but I'm including it in case we ever get dyeable towels.
+
+	// Washing allows you to remove all reagents from a towel, so it comes out clean!
+	reagents.remove_any(reagents.total_volume)
+
+	set_wet(FALSE, FALSE)
+	make_used(null, silent = TRUE)
+
+	var/fresh_mood = AddComponent(/datum/component/onwear_mood, saved_event_type = /datum/mood_event/fresh_laundry, examine_string = "[src] looks crisp and pristine.")
+
+	QDEL_IN(fresh_mood, 2 MINUTES)
+
+
+/obj/item/towel/dropped(mob/user, silent)
+	. = ..()
+
+	if(!ishuman(loc) && shape != TOWEL_FOLDED)
+		make_used(user, silent = TRUE)
+
+
+/**
+ * Helper proc to handle setting the towel's `wet` variable, changing the icon_state
+ * accordingly and making the towel look a little damp if wet, and removing that
+ * color if not wet.
+ *
+ * Arguments:
+ * * new_wetness - Whether we're now wet or not.
+ * * update_visuals (optional) - Whether we call `update_appearance()` and
+ * `update_slot_icon()`. Set to `FALSE` if you're already calling a proc that
+ * updates the towel's appearance, like `change_towel_shape()` (if you're sure
+ * that it WILL change the appearance). Defaults to `TRUE`.
+ */
+/obj/item/towel/proc/set_wet(new_wetness, update_visuals = TRUE)
+	if(new_wetness == wet)
+		return
+
+	wet = new_wetness
+
+	color = wet ? "#CCCCCC" : null
```

Comment:
maybe use `add_atom_color` instead?

Reply:
I don't think that's really making it better, I checked the code and it's mostly for when we plan on having multiple colors at once which don't appear at the same time, but rather based on priority, which isn't the case here.